classes/PHPTAL.php

```diff
@@ -791,7 +847,31 @@ public function executeMacro($path)
     private function setCodeFile()
     {
         $this->findTemplate();
-        $this->_codeFile = $this->getPhpCodeDestination() . $this->getFunctionName() . '.' . $this->getPhpCodeExtension();
+        $this->_codeFile = $this->getPhpCodeDestination() . $this->getSubPath() . $this->getFunctionName() . '.' . $this->getPhpCodeExtension();
+    }
+
+    /**
+     * Generate a subpath structure
+     *
+     * @return string
+     */
+    private function getSubPath()
```

Comment:
It would be great to provide a custom "CacheHandler", e.g. for saving the templates in memory caches. So we can implement our own caching strategies and logics for saving the files...